### PR TITLE
APIMF-2936: added cross spec restriction check. Moved it out of AMFCompiler

### PR DESCRIPTION
**Pull request by tomsfernandez**
Depends on: https://github.com/aml-org/amf-core/pull/179﻿

### CODE REVIEW

File: amf-webapi/shared/src/main/scala/amf/plugins/document/webapi/CrossSpecRestriction.scala [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+  protected def vendors: Seq[String]
+  protected def validVendorsToReference: Seq[Vendor] = Seq.empty
+
+  // TODO: all documents should have a Vendor
+  protected def restrictCrossSpecReferences(optionalReferencedVendor: Option[Vendor], reference: Reference)(
+      implicit errorHandler: ErrorHandler): Unit = {
+    val possibleReferencedVendors = vendors ++ validVendorsToReference.map(_.name)
+    optionalReferencedVendor.foreach { referencedVendor =>
+      if (!possibleReferencedVendors.contains(referencedVendor.name)) {
```
Comment (AgustinBettati) [inline amf-webapi/shared/src/main/scala/amf/plugins/document/webapi/CrossSpecRestriction.scala:19]: There used to be a check that allowed json references for all cases, im in favor of removing this custom check but do all plugins have this vendor set in the validVendorsToReference?

Comment (tomsfernandez) [reply amf-webapi/shared/src/main/scala/amf/plugins/document/webapi/CrossSpecRestriction.scala:19]: ExternalFragments doesn't have a vendor so it wouldn't apply for this condition. 

I think that the previous condition stated that an ExternalFragment could reference another document from another plugin. It was a hack for a cyclic reference case where ALS sent the ExternalFragment to parse and as it had a $ref it followed the parsing path. (I think)

**Review by looseale — DISMISSED** [review]

File: amf-webapi/shared/src/main/scala/amf/plugins/document/webapi/CrossSpecRestriction.scala (hunk 1 above)
Comment (looseale) [reply amf-webapi/shared/src/main/scala/amf/plugins/document/webapi/CrossSpecRestriction.scala:19]: It cames from this issue https://www.mulesoft.org/jira/browse/APIMF-2143 and it doesn't added a test. So we should check it to be sure that doesn't break anything.

Comment (tomsfernandez) [reply amf-webapi/shared/src/main/scala/amf/plugins/document/webapi/CrossSpecRestriction.scala:19]: Yup it has tests. 

See: https://github.com/aml-org/amf/blob/9ff718f5fb5418fe8ce6afa45f16399d7aeaf8ca/amf-client/shared/src/test/scala/amf/cycle/OasRecursiveFilesCycleTest.scala#L14-L53

**Review by looseale — APPROVED** [review]

**Pull request merged**